### DISCUSSION TITLE
[v5] How to add request logs

### DISCUSSION THREAD

**Discussion by yarapolana** [adonisjs/core#2843] [Help]
Hello, I am following [this guide](https://docs.adonisjs.com/guides/logger#using-pino-transports) to log requests in production.

I created a global middleware and here is what I have so far:

<img width="517" alt="Screenshot 2021-06-25 at 11 52 35" src="https://user-images.githubusercontent.com/19730118/123421516-780ad880-d5ac-11eb-9d5e-cd6beed28f18.png">

On the github of pino-sentry they have the stream inside the pino function but there is no stream option on the Logger, anyone know how to set this up. I am pretty new to this, not sure how to go around =) .

<img width="748" alt="Screenshot 2021-06-25 at 11 57 47" src="https://user-images.githubusercontent.com/19730118/123421601-92dd4d00-d5ac-11eb-928e-0f9c6e13d519.png">

**Comment by arthurcvm** [comment]
Hi @yarapolana,
please deconstruct `logger` in `HttpContextContract`, your function transform in someting like this:

```ts
public async handle ({ request, logger }: HttpContextContract... //Continuation of function 
```

and use this `logger` instance.

**Reply by yarapolana** [reply]
@arthurcvm that has the same result as the Logger, nowhere to add the stream.

**Reply by arthurcvm** [reply]
Sorry @yarapolana, I didn't understand your initial question and send a wrong answer.
In `config/app.ts` under `LoggerConfig` instance pass the `stream` attribute, like this:

```ts
export const logger: LoggerConfig = {
stream: Env.get('PINO_SENTRY'), //Continuation of object
```

**Reply by yarapolana** [reply]
@arthurcvm is there any other configuration missing, the server dies with this configuration.
```
Error: ENOENT: no such file or directory, open 'https://omitted@omitted.ingest.sentry.io/omitted'
Underlying HTTP server died with "1 code"
```

**Comment by thetutlage** [comment]
Yup as @arthurcvm said, you can override the stream property inside the config object. It will then send all the logs to Sentry

**Reply by yarapolana** [reply]
@thetutlage
Can you give me an example?
I am not sure I follow.

**Reply by yarapolana** [reply]
@thetutlage Ahhhhh my bad, I did not get @arthurcvm reply notification and was very confused.
This is all I needed. =) thank you both for the fast replies.
